### PR TITLE
[FLINK-24270][checkpoint] Refactor the tests related to the VertexFinishedStateChecker

### PR DESCRIPTION
**Pull request by gaoyunhaii**
## What is the purpose of the change

This PR refactors the tests related to the checking of restoring states for graph with finished vertices to directly tests the most atomic component, namely `VertexFinishedStateChecker`.

## Brief change log

- 68d4546ad8600dc12959a90aa338e3f91537db24 moved the related tests to a separate file and refactored the tests.

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): **no**
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: **no**
  - The serializers: **no**
  - The runtime per-record code paths (performance sensitive): **no**
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: **no**
  - The S3 file system connector: **no**

## Documentation

  - Does this pull request introduce a new feature? **no**
  - If yes, how is the feature documented? **not applicable**

### CODE REVIEW

**Comment by gaoyunhaii** [comment]
Hi @dawidwys @pnowojski could you have a look at this small refactor on test~?

File: flink-runtime/src/test/java/org/apache/flink/runtime/checkpoint/VertexFinishedStateCheckerTest.java [hunk 1]
```diff
@@ -0,0 +1,302 @@
...
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+import static java.util.Collections.singletonList;
+
+/** This tests verifies the checking logic of {@link VertexFinishedStateChecker}. */
+public class VertexFinishedStateCheckerTest {
+
+    @Rule public ExpectedException thrown = ExpectedException.none();
```
Comment (dawidwys) [inline flink-runtime/src/test/java/org/apache/flink/runtime/checkpoint/VertexFinishedStateCheckerTest.java:47]: Could we rewrite it with JUnit 5 already? I believe it should be as simple as changing the imports and switching to `assertThrows`. Example for `testRestoringPartiallyFinishedChainsFails`.

```
        final FlinkRuntimeException exception = assertThrows(
                FlinkRuntimeException.class,
                finishedStateChecker::validateOperatorsFinishedState);
        
        assertThat(exception.getMessage(), equalTo("Can not restore vertex "
                + "anon("
                + jobVertexID1
                + ")"
                + " which contain mixed operator finished state: [ALL_RUNNING, FULLY_FINISHED]"));
```

**Review by dawidwys — COMMENTED** [review]
Mostly, looks good.

Do we have a test that checks the `VertexFinishedStateChecker` is called while restoring? As per the description:

> It is probably a good idea to have two test against the Scheduler that validate that the modification tests happen at the right points. Something like testJobGraphModificationsAreCheckedForInitialSavepoint() and testJobGraphModificationsAreCheckedForInitialCheckpoint().

**Comment by gaoyunhaii** [comment]
Very thanks @dawidwys for the review! I change the junit usage and added the missed tests~

**Review by dawidwys — APPROVED** [review]
Thank you for the update! LGTM now

**Comment by gaoyunhaii** [comment]
Very thanks @dawidwys for the review!

**Pull request closed without merging**